diff --git a/lib/raop_handlers.h b/lib/raop_handlers.h
index 8dbd21b..b779ece 100644
--- a/lib/raop_handlers.h
+++ b/lib/raop_handlers.h
@@ -1129,11 +1129,11 @@
     if (!strcmp(content_type, "text/parameters")) {
         char *datastr = NULL;
         datastr = calloc(1, datalen + 1);
-        if (data && datastr && conn->raop_rtp) {
+        if (data && datastr) {
             memcpy(datastr, data, datalen);
             if ((datalen >= 8) && !strncmp(datastr, "volume: ", 8)) {
                 float vol = 0.0f;
                 sscanf(datastr+8, "%f", &vol);
-                if (raop_rtp_is_running(conn->raop_rtp)) {
+                if (conn->raop_rtp && raop_rtp_is_running(conn->raop_rtp)) {
                     raop_rtp_set_volume(conn->raop_rtp, vol);
                 } else if (raop->callbacks.audio_set_volume) {
@@ -1141,12 +1141,14 @@
                     raop->callbacks.audio_set_volume(raop->callbacks.cls, vol);
                 }
             } else if ((datalen >= 10) && !strncmp(datastr, "progress: ", 10)) {
-                uint32_t start = 0, curr = 0, end = 0;
-                sscanf(datastr+10, "%"PRIu32"/%"PRIu32"/%"PRIu32, &start, &curr, &end);
-                raop_rtp_set_progress(conn->raop_rtp, start, curr, end);
+                if (conn->raop_rtp) {
+                    uint32_t start = 0, curr = 0, end = 0;
+                    sscanf(datastr+10, "%"PRIu32"/%"PRIu32"/%"PRIu32, &start, &curr, &end);
+                    raop_rtp_set_progress(conn->raop_rtp, start, curr, end);
+                } else {
+                    logger_log(raop->logger, LOGGER_WARNING, "RAOP not initialized at SET_PARAMETER");
+                }
             }
-        } else if (!conn->raop_rtp) {
-            logger_log(raop->logger, LOGGER_WARNING, "RAOP not initialized at SET_PARAMETER");
         }
         free(datastr);
     } else if (!strcmp(content_type, "image/jpeg") || !strcmp(content_type, "image/png")) {
